OSSSpeechKit/Classes/OSSSpeech.swift

```diff
@@ -535,6 +544,40 @@ public class OSSSpeech: NSObject {
             delegate?.didFailToCommenceSpeechRecording()
             delegate?.didFailToProcessRequest(withError: OSSSpeechKitErrorType.invalidSpeechRequest.error)
         }
+        
+        readyToRecord()
+    }
+    
+    func readyToRecord()
+    {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
+        let dateString = dateFormatter.string(from: Date())
+        
+        self.audioFileURL = getDocumentsDirectory().appendingPathComponent("\(dateString).m4a")
+
+        let audioSettings = [AVFormatIDKey: Int(kAudioFormatMPEG4AAC),
+                           AVSampleRateKey: 12000,
+                     AVNumberOfChannelsKey: 1,
+                  AVEncoderAudioQualityKey: AVAudioQuality.high.rawValue
+        ]
+        
+        do {
+            audioRecorder = try AVAudioRecorder(url: audioFileURL, settings: audioSettings)
+            audioRecorder?.delegate = self
+            audioRecorder?.record()
+
+        }
+        catch
+        {
+            print(error.localizedDescription)
```

Comment:
We should add a new `OSSSpeechKitErrorType` and then use the `didFailToProcessRequest(withError` function to handle sharing the error.